lib/posix/semaphore.c

```diff
@@ -35,7 +35,13 @@ int sem_destroy(sem_t *semaphore)
  */
 int sem_getvalue(sem_t *semaphore, int *value)
 {
+	if (!semaphore) {
```

Comment:
please check for NULL here and make If statement have essentially Boolean type

see 4218d5f8f04795c9508c36d51b0b1179ba57b7c4

Reply:
@nashif pushed changes. Can you please have a look ?